index.js

```diff
@@ -1,30 +1,162 @@
-main(input, 30);
+
+const consume = async (msgHandler, errorHandler) => {
+    let message = await getMessage();
+    if (!message) {
+        console.log('No messages in queue');
+        consume(msgHandler, errorHandler);
+    } else {
+        try {
+            await msgHandler(message);
+        } catch (err) {
+            errorHandler(err);
+        }
+        consume(msgHandler, errorHandler);
+    }
+};
+
+const messageHandler = async (msg) => {
+    if (!isValidMessage(msg)) throw new Error('Invalid message from queue');
+    const msgBody = JSON.parse(msg.Body);
+    const prompt = promptBuilder(msgBody);
+
+    let flashcards;
+    try {
+        const response = await generateWithRetries(msgBody.inputText, prompt);
+        if (!isValidFlashcardResponse(response)) {
+            throw new Error('Invalid flashcard response from API');
+        }
+        flashcards = JSON.parse(response).flashcards;
+
+        if (flashcards.length < msgBody.cardCount) {
+            throw new Error('Response has too few flashcards');
+        }
+        if (flashcards.length > msgBody.cardCount) {
+            flashcards = flashcards.slice(0, msgBody.cardCount);
+        }
+    } catch (err) {
```

Comment:
nit: why would you wrap error thrown above immediately with another error? the only unhandled error is 
`const response = await generateWithRetries(msgBody.inputText, prompt);` and perhaps `JSON.parse`
I would just `try catch` these two lines separately and give it also distinct error message